### PR TITLE
Add the sdk/internal/x package

### LINKED ISSUE #5436: Add an sdk/internal/x package
The following detectors are need to be added to comply with semantic conventions.

- https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5399
- https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5424

However, these detectors overlap with existing host and service detectors in this repository but they contain experimental semantics. To support these experimental semantics, we plan to extend the existing host and service resource detectors. This extension will be gated with an environment variable.

Similar to [go.opentelemetry.io/otel/sdk/metric/internal/x](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric/internal/x),  add an `x` package to facilitate this..

### PR DESCRIPTION
**Pull request by MrAlias**
Add the `x` package to handle experimental feature flagging within the go.opentelemetry.io/otel/sdk module. Currently this only supports enabling experimental resource semantic conventions.

Resolve #5436 

cc @pyohannes

### CODE REVIEW

**Review by dashpole — APPROVED** [review]

File: sdk/internal/x/x.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+//
+// To enable this feature set the OTEL_GO_X_RESOURCE environment variable
+// to the case-insensitive string value of "true" (i.e. "True" and "TRUE"
+// will also enable this).
+var Resource = newFeature("RESOURCE", func(v string) (string, bool) {
+	if strings.ToLower(v) == "true" {
+		return v, true
+	}
+	return "", false
+})
```
Comment (XSAM) [inline sdk/internal/x/x.go:26]: Should we follow this guideline to warn our users when the fallback happens?

> If any value other than a true value, case-insensitive string "false", empty, or unset is used, a warning SHOULD be logged to inform users about the fallback to false being applied.

https://github.com/open-telemetry/opentelemetry-specification/blob/62effed618589a0bec416a87e559c0a9d96289bb/specification/configuration/sdk-environment-variables.md#boolean-value

Probably not in this PR, we can create an issue to track this.

**Review by XSAM — APPROVED** [review]

File: sdk/internal/x/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+[OpenTelemetry resource semantic conventions]: https://opentelemetry.io/docs/specs/semconv/resource/
+[resource detectors]: https://pkg.go.dev/go.opentelemetry.io/otel/sdk/resource#Detector
+
+#### Examples
+
+Enable experimental resource semantic conventions.
+
+```console
+export OTEL_GO_X_RESOURCE=true
```
Comment (pyohannes) [inline sdk/internal/x/README.md:30]: Should there be this single environment variable which turns on and off _all_ experimental semantic conventions, or is it the intention to have something more specific? For example, have a `OTEL_GO_X_RESOURCE_SERVICE`, which enables experimental attributes on the service resource?

**Review by pyohannes — APPROVED** [review]
Thanks @MrAlias for making this available.

File: sdk/internal/x/README.md (hunk 2 above)
Comment (MrAlias) [reply sdk/internal/x/README.md:30]: I don't expect there is a use case where a user wants some but not all of the experimental environment variables. 

If we find that to exist, we could add specific variables to support that then.

What do you think?

File: sdk/internal/x/x.go (hunk 1 above)
Comment (MrAlias) [reply sdk/internal/x/x.go:26]: Sure. We would want to address this in all places we support boolean environment variables though.

Comment (MrAlias) [reply sdk/internal/x/x.go:26]: Tracking [here](https://github.com/open-telemetry/opentelemetry-go/issues/5471).

File: sdk/internal/x/README.md (hunk 2 above)
Comment (pyohannes) [reply sdk/internal/x/README.md:30]: Sounds good. Once this is merged, I'll submit a PR for `service.instance.id` guarded by the `Resource` flag.

**Pull request merged**